platform/android/src/style/sources/sources.cpp

```diff
@@ -43,11 +45,16 @@ namespace android {
 jni::jobject* createJavaSourcePeer(jni::JNIEnv& env, AndroidRendererFrontend& frontend, mbgl::style::Source& coreSource) {
     std::unique_ptr<Source> peerSource = std::unique_ptr<Source>(initializeSourcePeer(coreSource));
```

Comment:
This doesn't look right. Why construct a unique pointer from a raw pointer, which might be owned by some other `unique_ptr`, and then release it later?